lib/session.js

```diff
@@ -20,26 +19,29 @@ const hash = Symbol('hash')
 
 module.exports = class Session {
   constructor (request, idGenerator, cookieOpts, secret, prevSession = {}) {
+    // Copy over values from the previous session
+    for (const key in prevSession) {
+      this[key] = prevSession[key]
+    }
+
+    this[requestKey] = request
     this[generateId] = idGenerator
-    this.expires = null
-    this.cookie = new Cookie(cookieOpts)
     this[cookieOptsKey] = cookieOpts
-    this[maxAge] = cookieOpts.maxAge
     this[secretKey] = secret
-    this[addDataToSession](prevSession)
```

Comment:
I think this is extremely unreadable and buried. It caused me a lot of confusion earlier, so I decided to put in the function itself